src/queryexecutor.js

```diff
@@ -0,0 +1,350 @@
+          break;
+        }
+        if (q.startsWith('query ')) {
+          q = query.slice(2).join(' ');
+          redirectTo_(`https://dune.xyz/browse/queries?q=${encodeURIComponent(q)}`);
+          break;
+        }
+        redirectTo_(`https://dune.xyz/browse/dashboards?q=${encodeURIComponent(q)}`);
+        break;
+      }
+
+      case "ipfs": {
+        const cid = query[1];
+        if (!cid) {
+          redirectTo_('https://app.pinata.cloud/');
+          break;
+        }
+        redirectTo_(`https://gateway.ipfs.io/ipfs/${cid}`);
+        break;
+      }
+
+      case "ipns": {
+        const url = query[1];
+        if (!url) {
+          redirectTo_('https://app.pinata.cloud/');
+          break;
+        }
+        redirectTo_(`https://gateway.ipfs.io/ipns/${url}`);
+        break;
+      }
+
+      case "arweave":
+      case "ar": {
+        const cid = query[1];
+        if (!cid) {
+          redirectTo_('https://viewblock.io/arweave');
+          break;
+        }
+        redirectTo_(`https://arweave.net/${cid}`);
+        break;
+      }
+
+      case "gm":
+      case "gmi":
+      case "wagmi": {
+        redirectTo_('./gm.html');
+        break;
+      }
+
+      case "ngmi": {
+        const ngmis = [
+          'https://meta.com/',
+          'https://www.sec.gov/',
+          'https://www.alphabet.com/',
+        ];
+        redirectTo_(ngmis[Math.floor(Math.random()*ngmis.length)]);
+        break;
+      }
+
+      case "elon": {
+        const elon = [
+          '1392030108274159619',
+          '1375033483148451842',
+          '1387290679794089986',
+          '1395027147161489412',
+          '1454876031232380928',
+          '1451015695106560000',
+          '1426558939336937481'
+        ];
+        const id = elon[Math.floor(Math.random()*elon.length)];
+        redirectTo_(`https://twitter.com/elonmusk/status/${id}`);
+        break;
+      }
+
+      default: {
+        if (query.length === 1 && query[0].endsWith('.eth')) {
+          redirectTo_(`https://etherscan.io/address/${commandName}`);
+        } else {
+          redirectTo_(`http://www.google.com/search?q=${encodeURIComponent(query.join(" "))}`);
+        }
+      }
+    }
+  }
+}
+
+function redirectTo_(url) {
+  window.location = url;
+}
+
+function getQueryParams_() {
+  // First, check if the command is in the query params
+  const searchParams = new URLSearchParams(window.location.search);
+  const searchQuery = searchParams.get("q");
+  if (searchQuery) {
+    return searchQuery.split(" ");
+  }
+  // Next, check if it's in the hash params
+  const hashParams = new URLSearchParams((window.location.hash ?? '').split('?')[1]);
+  const hashQuery = hashParams.get("q");
+  if (hashQuery) {
+    return hashQuery.split(" ");
+  }
+  // Otherwise, there is no command set
+  return ["homepage"];
+}
+
+// Top 100 CoinGecko coins. Hardcoded to provide fast lookups.
+const COIN_GECKO_SYMBOL_MAP = {
```

Comment:
nit, we could extract this out

Reply:
yeah I had whole another changes that refactored granular stuff out but did not know if it was ok to take dependency on web pack, we can do it after